sound/soc/sof/core.c

```diff
@@ -131,19 +131,6 @@ struct snd_sof_dai *snd_sof_find_dai(struct snd_sof_dev *sdev,
 	return NULL;
 }
 
-bool snd_sof_dsp_d0i3_on_suspend(struct snd_sof_dev *sdev)
-{
-	struct snd_sof_pcm *spcm;
-
-	list_for_each_entry(spcm, &sdev->pcm_list, list) {
-		if (spcm->stream[SNDRV_PCM_STREAM_PLAYBACK].suspend_ignored ||
-		    spcm->stream[SNDRV_PCM_STREAM_CAPTURE].suspend_ignored)
-			return true;
-	}
-
-	return false;
-}
-
```

Comment:
help me out on this one: weren't you the one who suggested using a helper instead of having yet another flag for the state machine? Why would we want to remove the helper here?

Reply:
@plbossart I knew I's get this question. Unfortunately with the multi-client setup, the SOF top-level device is oblivious to the list of PCM's as they will only be available to the client device. So when the client decides to keep a D0i3_compatible pipeline running, it has to notify the SOF device of the decision.

Reply:
ok, but is using a flag the right direction then? 
If you have two clients, we probably don't want direct access to the d0i3_on_suspend field.

Reply:
@plbossart Yes, this is the tricky part. I think I have rework the S0ix parts first before I address the problem in my PR. I am working on it.